Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -274,7 +278,7 @@ namespace mu2e {
               KKCALOHITCOL addcalohits;
               KKSTRAWXINGCOL addstrawxings;
               kkfit_.addStrawHits(*tracker, *strawresponse, *kkbf_, kkmat_.strawMaterial(), *kktrk, chcol, addstrawhits, addstrawxings );
-              if(kkfit_.useCalo())kkfit_.addCaloHit(*calo_h, *kktrk, cc_H, addcalohits);
+              if(kkfit_.useCalo()&&calohits.size()==0)kkfit_.addCaloHit(*calo_h, *kktrk, cc_H, addcalohits);
```

Comment:
I am not an expert on this code but I thought I would check one thing - did you mean size()!=0?